File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let numArr=[];
    let node=head;
    while(node){
        numArr.push(node.val);
        node=node.next;
    }
    head=new ListNode(0,head);
    let prev=head,next=head.next;
    let num= numArr.length-n+1;
    
    while(next){
          if(numArr.length===n){
              prev.next=next.next;
          }else{
              prev=prev.next;
          }
        numArr.pop();
        next=next.next;
        
    }
    return head.next;
};

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        lol=head
        i=0
        while lol:
            lol=lol.next
            i+=1
        ni=i-n
        if ni==0:
            return head.next
        lol=head
        for i in range(ni-1):
            lol=lol.next
        lol.next=lol.next.next
        return head